extensions/src/main/java/org/apache/falcon/extensions/store/ExtensionStore.java

```diff
@@ -346,4 +347,32 @@ public boolean isExtensionStoreInitialized() {
         return (storePath != null);
     }
 
+    public String disableExtension(final String extensionName, String currentUser) throws FalconException {
+        ExtensionType extensionType = AbstractExtension.isExtensionTrusted(extensionName) ? ExtensionType.TRUSTED
+                : ExtensionType.CUSTOM;
+        if (extensionType.equals(ExtensionType.TRUSTED)) {
+            throw new ValidationException(extensionName + " is trusted cannot be disabled.");
+        } else if (!metaStore.checkIfExtensionExists(extensionName)) {
+            throw new FalconException("Extension:" + extensionName + " is not registered with Falcon.");
+        } else if (!metaStore.getDetail(extensionName).getExtensionOwner().equals(currentUser)) {
+            throw new FalconException("User: " + currentUser + " is not allowed to disable extension: "
+                    + extensionName);
+        } else {
```

Comment:
Do you also want to check if it is already disabled and show an appropriate message?